In [1]:
import pandas as pd

In [5]:
news_read1 = pd.read_csv('../../selenium/yohan/crawling/news_get.daum_news.csv')
news_read2 = pd.read_csv('../../selenium/sky/crawling/local.daum_news.csv')
news_read3 = pd.read_csv('../../selenium/juhyeon/crawling/daum_news_2020-2021.csv')

df = [news_read1,news_read2,news_read3]

In [6]:
df_news = pd.concat(df)
df_news

_id                                 news_title  \
0     65f273a3cce44e8d1d610256                   [재송] 31일 장 마감 후 주요 종목 뉴스   
1     65f273aacce44e8d1d610257                "고전게임 아냐?"..메타버스 시대, 2D가 온다   
2     65f273b3cce44e8d1d610258  미래기술 경연장 美CES 금주 개막..韓, 역대 최대 400여개 기업 참가   
3     65f273bacce44e8d1d610259       'CES 2022' 금주 개막..韓 400개 기업 미래기술 선뵌다   
4     65f273c1cce44e8d1d61025a   美 CES 2년만에 오프라인 개막.."韓 역대 최다 500개 기업 참가"   
...                        ...                                        ...   
8431  65f3c6ef155e2f313db29482                     한컴MDS, 이창열 대표이사 체제로 변경   
8432  65f3c6f0155e2f313db29483                  한컴MDS, 이창열 대표이사 단독 체제로 운영   
8433  65f3c6f2155e2f313db29484          '칠곡할매글꼴 문화유산 등재'..군민이 뽑은 10대뉴스 1위   
8434  65f3c6f3155e2f313db29485        CES 2022 관전 포인트 '스마트홈·디지털헬스·로봇·NFT'   
8435  65f3c6f4155e2f313db29486                        31일 장 마감 후 주요 종목 뉴스   

                news_date news_brand  \
0       2022. 1. 1. 06:00       이데일리   
1       2022. 1. 2. 09:00        뉴스1   
2       2022. 1. 2. 15:55       연합뉴스   
3       2022. 1. 2. 16:15     한국경제TV   
4       2022. 1. 2. 16:32       조선비즈   
...                   ...        ...   
8431  2021. 12. 31. 10:14        뉴시스   
8432  2021. 12. 31. 10:38       이데일리   
8433  2021. 12. 31. 16:45        뉴스1   
8434    2022. 1. 1. 11:38     지디넷코리아   
8435  2021. 12. 31. 18:31       이데일리   

                                          news_contents  
0     [이데일리 이대호 기자] 다음은 31일 장 마감 이후 주요 종목 뉴스다.\n△대웅제...  
1     네이버제트, 두나무, 싸이월드 등 2D 기반 메타버스 서비스 속속\n"구현·활용 쉽...  
2     삼성, 韓기업 유일하게 기조연설..정의선 현대차 회장, 직접 현장 찾아\nLG·SK...  
3     [한국경제TV 김현경 기자]\n세계 최대 전자·IT 전시회인 'CES 2022'가 ...  
4     SK그룹이 오는 5일(현지시각) 열리는 세계 최대 전자·정보기술(IT) 전시회인 C...  
...                                                 ...  
8431  [서울=뉴시스]신항섭 기자 = 한컴MDS는 전동욱 대표이사 사임으로 이창열 대표이사...  
8432  [이데일리 양지윤 기자] 한컴MDS(086960)는 31일 전동욱 대표이사가 일신상...  
8433  (칠곡=뉴스1) 정우용 기자 = 경북 칠곡군은 31일 주민, 네티즌 등 2000여명...  
8434  (지디넷코리아=이나리 기자)세계 최대 가전전시회인 'CES 2022'가 새 해 1월...  
8435  [이데일리 이대호 기자] 다음은 31일 장 마감 이후 주요 종목 뉴스다. △대웅제약...  

[18508 rows x 5 columns]

In [8]:
# 여러번에 걸쳐 수집이 되었으므로 중복값을 드랍
df_news = df_news.drop(columns='_id')
df_news = df_news.drop_duplicates()
df_news

news_title            news_date  \
0                      [재송] 31일 장 마감 후 주요 종목 뉴스    2022. 1. 1. 06:00   
1                   "고전게임 아냐?"..메타버스 시대, 2D가 온다    2022. 1. 2. 09:00   
2     미래기술 경연장 美CES 금주 개막..韓, 역대 최대 400여개 기업 참가    2022. 1. 2. 15:55   
3          'CES 2022' 금주 개막..韓 400개 기업 미래기술 선뵌다    2022. 1. 2. 16:15   
4      美 CES 2년만에 오프라인 개막.."韓 역대 최다 500개 기업 참가"    2022. 1. 2. 16:32   
...                                         ...                  ...   
8311               한컴위드, NFT 거래 플랫폼 '한컴아트피아' 출시  2021. 12. 20. 10:58   
8327       한글과컴퓨터 "싸이월드 메타버스 공간 '한컴타운'서 이벤트 개최"  2021. 12. 22. 09:47   
8346          '스우파' 허니제이, 싸이월드 메타버스서 팬들과 특별한 만남  2021. 12. 23. 08:50   
8378           국내 첫 민간 지구관측용 위성 '세종1호' 내년 6월 발사  2021. 12. 23. 18:47   
8392     한컴그룹, 성균관대·한예종 등에 '미래인재 육성' 장학금 2억원 기탁  2021. 12. 27. 10:22   

     news_brand                                      news_contents  
0          이데일리  [이데일리 이대호 기자] 다음은 31일 장 마감 이후 주요 종목 뉴스다.\n△대웅제...  
1           뉴스1  네이버제트, 두나무, 싸이월드 등 2D 기반 메타버스 서비스 속속\n"구현·활용 쉽...  
2          연합뉴스  삼성, 韓기업 유일하게 기조연설..정의선 현대차 회장, 직접 현장 찾아\nLG·SK...  
3        한국경제TV  [한국경제TV 김현경 기자]\n세계 최대 전자·IT 전시회인 'CES 2022'가 ...  
4          조선비즈  SK그룹이 오는 5일(현지시각) 열리는 세계 최대 전자·정보기술(IT) 전시회인 C...  
...         ...                                                ...  
8311      아시아경제  [아시아경제 강나훔 기자] 한컴위드는 대체불가능토큰(NFT) 거래 플랫폼 '한컴아트...  
8327        뉴시스  [서울=뉴시스] 김경택 기자 = 한글과컴퓨터는 싸이월드의 메타버스 공간 '싸이월드 ...  
8346        뉴스엔  [뉴스엔 이민지 기자] 훌리뱅 리더 허니제이가 싸이월드 메타버스 공간에서 팬들과 만...  
8378        KBS  한글과컴퓨터그룹이 국내 첫 민간 지구 관측용 위성인 '세종1호(Sejong-1)'를...  
8392      머니투데이  한글과컴퓨터(한컴)그룹이 미래 인재 육성을 위해 총 2억원 상당의 장학금을 국내 주...  

[17586 rows x 4 columns]

In [12]:
# 뉴스 브랜드 확인
df_news['news_brand'].unique()

array(['이데일리', '뉴스1', '연합뉴스', '한국경제TV', '조선비즈', '노컷뉴스', '서울경제', '쿠키뉴스',
       '뉴시스', '아시아경제', '머니투데이', '아이뉴스24', '전자신문', '파이낸셜뉴스', '지디넷코리아',
       '매경이코노미', '한국경제', '일간스포츠', '세계일보', '디지털타임스', 'IT동아', '동아사이언스',
       '헤럴드경제', '스포츠조선', '게임톡', '조선일보', '경향신문', '동아일보', 'SB', '중앙SUNDAY',
       '비즈워치', '한국일보', '매일경', '중앙일보', '머니', '데일리안', '한경비즈니스', '스포츠경향',
       '여성동아', '더팩트', '프레시안', '오마이뉴스', '정책브리핑', '서울신문', '우먼센스', 'MBN',
       '스포티비뉴스', '뉴스엔', '게임동아', '스포츠서울', '대전일보', '포모스', '데일리e스포츠', 'YTN',
       '국민일보', '시사저널', '문화일보', '국제신문', '매경게임진', 'KBS', '한겨레', 'JTBC',
       'OSEN', '스타뉴스', '스타투데이', '이코노미조선', '코리아헤럴드', '연합뉴스TV', '스포츠동아',
       '엑스포츠뉴스', 'MK스포츠', '코리아중앙데일리', '주간동아', '스포츠한국', '코스모폴리탄', '미디어오늘',
       'MBC&iMB', 'Copyrigh', '마이데일리', '헬스조선', '코메디닷컴', '더스쿠프', '베스트일레븐',
       '대구MBC', 'TV리포트', 'bnt뉴스', '경기일보', '로피시', 'kbc광주방송', 'MHN스포츠',
       '스포츠투데이', '스포탈코리아', '트래비', '매일경제', '머니S', '시티라이프', '채널A', '티브이데일리',
       'SBS연예뉴스', 'SBS', 'MBC', '아레나', '베이비뉴스', '주간경향', '키즈맘', '

- 이상치로 '매일경', '머니', 'Copyrigh' 항목이 존재
- '매일경' = '매일경제', '머니' = '머니투데이', 'Copyrigh' = '전자신문' 으로 변경

In [19]:
df_news.loc[df_news['news_brand'] == '매일경', 'news_brand'] = '매일경제'
df_news.loc[df_news['news_brand'] == '머니', 'news_brand'] = '머니투데이'
df_news.loc[df_news['news_brand'] == 'Copyrigh', 'news_brand'] = '전자신문'


In [24]:
df_news.info()

<class 'pandas.core.frame.DataFrame'>
Index: 17586 entries, 0 to 8392
Data columns (total 4 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   news_title     17586 non-null  object
 1   news_date      17396 non-null  object
 2   news_brand     17586 non-null  object
 3   news_contents  17584 non-null  object
dtypes: object(4)
memory usage: 1.2+ MB


- date값이 object타입인것을 확인 -> date타입으로 변경
- 크롤링한 주식의 단위가 일일 단위이므로 시간은 삭제

In [27]:
df_news['news_date'] = pd.to_datetime(df_news['news_date']).dt.date

In [28]:
df_news.info()

<class 'pandas.core.frame.DataFrame'>
Index: 17586 entries, 0 to 8392
Data columns (total 4 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   news_title     17586 non-null  object
 1   news_date      17396 non-null  object
 2   news_brand     17586 non-null  object
 3   news_contents  17584 non-null  object
dtypes: object(4)
memory usage: 1.2+ MB


- 총 row의 갯수는 17586개
- date와 contents에서 Null값이 확인되고 있으므로 해당 row확인 작업진행

In [30]:
df_news.query('news_contents.isna()')

news_title   news_date news_brand news_contents
343     <사진>한컴그룹, 서울노인복지센터서 급식 봉사활동 펼쳐  2018-05-14     디지털타임스           NaN
7221  싸늘하게 식은 홍콩 IPO 시장..올 3분기 한국이 제쳤다  2021-10-17      머니투데이           NaN

In [31]:
df_news.query('news_date.isna()')

news_title news_date news_brand  \
49                            정부 지원·대기업 러브콜..엔진 달구는 로봇주       NaT      헤럴드경제   
113                     김상철 한컴그룹 회장 "인공위성 5기 빠른 시일내 발사"       NaT      헤럴드경제   
141                    삼성증권, 판교금융센터 개점식 메타버스로 진행..업계 최초       NaT      헤럴드경제   
144           CES 2022로 본 미래 기술..'차세대 모빌리티'와 '로봇'을 주목하라       NaT     한경비즈니스   
180                        한컴인텔리전스, 디지털 트윈기술 기업 스탠스 품었다       NaT       한국경제   
...                                                 ...       ...        ...   
3763                              오리콤, 새 CEO에 박병철 사장 선임       NaT      헤럴드경제   
3790      한컴어썸텍, 국방기술진흥연구소·대전시 공동 운영 '대전국방벤처센터' 협약기업 발탁       NaT   중앙SUNDAY   
3830                한컴 김상철회장 아들 구속에…김연수 한컴 대표 “누나로서 송구”       NaT      헤럴드경제   
3930                    한컴라이프케어, 방위사업청과 115억 규모 공급계약 체결       NaT      헤럴드경제   
3931  한컴라이프케어 수주공시 - K5방독면 7차 양산사업 115.4억원 (매출액대비 11...       NaT       한국경제   

                                          news_contents  
49    로보티즈·로보로보·유진로봇 등\n최근 한달새 100% 이상씩 급등\nCES서도 두각...  
113   김상철 한컴그룹 회장이 7일(현지시간) 미국 라스베이거스 CES2022 한글과컴퓨터...  
141   삼성증권 제공\n[헤럴드경제=양대근 기자] 삼성증권은 판교금융센터 오픈을 기념해 증...  
144   [테크 트렌드]\n\n정의선 현대자동차그룹 회장이 1월 4일(현지시간) 미국 라스베...  
180   이 기사는 01월 19일 15:32 “마켓인사이트”에 게재된 기사입니다.\n한컴MD...  
...                                                 ...  
3763  박병철 오리콤 신임 CEO [오리콤 제공]\n[헤럴드경제=김민지 기자] 종합광고회사...  
3790  한컴어썸텍, 국방기술진흥연구소·대전시 공동 운영 '대전국방벤처센터' 협약기업 발탁\...  
3830  김연수 한컴 대표이사. [박혜림 기자/rim@]\n[헤럴드경제=박세정 기자] 한글과...  
3930  [헤럴드경제=증권부] 한컴라이프케어는 방위사업청과 K5방독면 7차 양산산업 계약을 ...  
3931  12월 14일 한컴라이프케어(372910)는 수주공시를 발표했다.\n◆한컴라이프케어...  

[190 rows x 4 columns]

- 총 갯수가 매우 적으므로 배제하고 진행
    + 날짜의 경우 인덱스번호로 유추할 수 있으나 정확하지 않으므로 사용하지 않음

In [32]:
df_news = df_news.dropna()
df_news

news_title   news_date news_brand  \
0                      [재송] 31일 장 마감 후 주요 종목 뉴스  2022-01-01       이데일리   
1                   "고전게임 아냐?"..메타버스 시대, 2D가 온다  2022-01-02        뉴스1   
2     미래기술 경연장 美CES 금주 개막..韓, 역대 최대 400여개 기업 참가  2022-01-02       연합뉴스   
3          'CES 2022' 금주 개막..韓 400개 기업 미래기술 선뵌다  2022-01-02     한국경제TV   
4      美 CES 2년만에 오프라인 개막.."韓 역대 최다 500개 기업 참가"  2022-01-02       조선비즈   
...                                         ...         ...        ...   
8311               한컴위드, NFT 거래 플랫폼 '한컴아트피아' 출시  2021-12-20      아시아경제   
8327       한글과컴퓨터 "싸이월드 메타버스 공간 '한컴타운'서 이벤트 개최"  2021-12-22        뉴시스   
8346          '스우파' 허니제이, 싸이월드 메타버스서 팬들과 특별한 만남  2021-12-23        뉴스엔   
8378           국내 첫 민간 지구관측용 위성 '세종1호' 내년 6월 발사  2021-12-23        KBS   
8392     한컴그룹, 성균관대·한예종 등에 '미래인재 육성' 장학금 2억원 기탁  2021-12-27      머니투데이   

                                          news_contents  
0     [이데일리 이대호 기자] 다음은 31일 장 마감 이후 주요 종목 뉴스다.\n△대웅제...  
1     네이버제트, 두나무, 싸이월드 등 2D 기반 메타버스 서비스 속속\n"구현·활용 쉽...  
2     삼성, 韓기업 유일하게 기조연설..정의선 현대차 회장, 직접 현장 찾아\nLG·SK...  
3     [한국경제TV 김현경 기자]\n세계 최대 전자·IT 전시회인 'CES 2022'가 ...  
4     SK그룹이 오는 5일(현지시각) 열리는 세계 최대 전자·정보기술(IT) 전시회인 C...  
...                                                 ...  
8311  [아시아경제 강나훔 기자] 한컴위드는 대체불가능토큰(NFT) 거래 플랫폼 '한컴아트...  
8327  [서울=뉴시스] 김경택 기자 = 한글과컴퓨터는 싸이월드의 메타버스 공간 '싸이월드 ...  
8346  [뉴스엔 이민지 기자] 훌리뱅 리더 허니제이가 싸이월드 메타버스 공간에서 팬들과 만...  
8378  한글과컴퓨터그룹이 국내 첫 민간 지구 관측용 위성인 '세종1호(Sejong-1)'를...  
8392  한글과컴퓨터(한컴)그룹이 미래 인재 육성을 위해 총 2억원 상당의 장학금을 국내 주...  

[17394 rows x 4 columns]